wgengine/packet/packet.go

```diff
@@ -279,17 +242,47 @@ func (q *QDecode) Decode(b []byte) {
 	}
 }
 
-// Returns a subset of the IP subprotocol section.
+func (q *QDecode) IPHeader() IPHeader {
+	ipid := get16(q.b[4:6])
+	return IPHeader{
+		IPID:    ipid,
+		IPProto: q.IPProto,
+		SrcIP:   q.SrcIP,
+		DstIP:   q.DstIP,
+	}
+}
+
+func (q *QDecode) ICMPHeader() ICMPHeader {
+	return ICMPHeader{
+		IPHeader: q.IPHeader(),
+		Type:     ICMPType(q.b[q.subofs+0]),
+		Code:     ICMPCode(q.b[q.subofs+1]),
+	}
+}
+
+func (q *QDecode) UDPHeader() UDPHeader {
+	return UDPHeader{
+		IPHeader: q.IPHeader(),
+		SrcPort:  q.SrcPort,
+		DstPort:  q.DstPort,
+	}
+}
+
+// Returns the IP subprotocol section.
```

Comment:
// Sub returns the ...

(complete sentence w/ subject)

Likewise elsewhere in this PR.